website/profile/views.py

```diff
@@ -450,18 +450,22 @@ def user_choose_mailing_lists(auth, **kwargs):
     json_data = escape_html(request.get_json())
     if json_data:
         for list_name, subscribe in json_data.items():
-            update_subscription(user, list_name, subscribe)
+            # TO DO: change this to take in any potential non-mailchimp, something like try: update_subscription(), except IndexNotFound: update_mailchimp_subscription()
+            if list_name == 'Open Science Framework Help':
+                help_mails(user=user, subscribe=subscribe)
+            else:
+                update_mailchimp_subscription(user, list_name, subscribe)
     else:
         raise HTTPError(http.BAD_REQUEST, data=dict(
             message_long="Must provide a dictionary of the format {'mailing list name': Boolean}")
         )
 
     user.save()
-    return {'message': 'Successfully updated mailing lists', 'result': user.mailing_lists}, 200
+    return {'message': 'Successfully updated mailing lists', 'result': user.mailchimp_mailing_lists.update(user.osf_mailing_lists)}, 200
```

Comment:
I believe `user.mailchimp_mailing_lists.update(user.osf_mailing_lists)` just updates the dictionary without returning the modified value. Also, instead of updating `user.mailchimp_mailing_lists` directly, consider saving it to a `result` variable and updating that so the mailing list record is not accidentally altered.